text/3850-autogenerated-attr.md

```diff
@@ -0,0 +1,121 @@
+- Feature Name: autogenerated_attr
+- Start Date: 2025-08-24
+- RFC PR: [rust-lang/rfcs#3850](https://github.com/rust-lang/rfcs/pull/3850)
+- Rust Issue: [rust-lang/rust#0000](https://github.com/rust-lang/rust/issues/0000)
+
+# Summary
+[summary]: #summary
+
+Add an attribute for marking items as being generated by a tool.
+```rust
+mod foo {
+	#![autogenerated(
+		by = "example tool",
+		suppress_lints = false,
+		format = true,
+		regen_hint = "./regen_example_tool",
+		document = false,
+		stable_api = true,
+	)]
+	// autogenerated code here...
+}
+```
+Many attributes could be provided to allow easy configuration of what would break code and what wouldn't. For example, if someone is using a test and string manipulation to change code at test time, then the tool likely would need `format = false`.
+
+# Motivation
+[motivation]: #motivation
+
+Rust doesn't have a general attribute for this (some more specific attributes do exist, such as `#[automatically_derived]`), and this would be useful for scenarios such as:
+- `rust-analyzer`: Mark files as autogenerated and discourage users from editing them
+- `rustfmt`: Don't format files marked as autogenerated by default
```

Comment:
Auto-generated code should get formatting to be readable to humans, because the generator sure doesn't need to duplicate all formatting logic into itself.